Wootric/WootricSDK-Android
# Add custom scale support

Add support for custom scale

## Changes
- Refactor Score to store array of score rules for each type
- Update Settings to keep the surveyTypeScale
- Update tests

## Test

1. Checkout this branch
2. Be sure to test with a `CSAT` account (the only mode that has custom scales)
3. Set the survey type scale
```java
        Wootric wootric = Wootric.init(this, CLIENT_ID, ACCOUNT_TOKEN);
        wootric.setEndUserEmail("nps@example.com");
        wootric.setSurveyTypeScale(1);
        wootric.setSurveyImmediately(true);
        wootric.survey();
```
4. Test for both iPhone & iPad
5. If wrong number is passed (e.g. -1, 4, 20) it should return default scale for that mode.
6. Check that `sdk_version` has following format: `android-X.Y.Z`

**Internal Trello Ref:** [link](https://trello.com/c/qDKxZtWB/4998-5-android-sdk-customcsat-scale-for-hootsuite-3)
